src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -824,7 +824,7 @@ void EmitFixedSet_LeftToRight()
 
                         if (setsToUse > 1)
                         {
-                            using (EmitBlock(writer, $"if (i >= span.Length - {minRequiredLength - 1})"))
+                            using (EmitBlock(writer, $"if ((uint)(i + {minRequiredLength - 1}) >= span.Length)"))
```

Comment:
NIT: since this check is technically a little harder to read than the original, can we emit a comment above this? No need to explain why casting to uint, I was thinking that perhaps just a simple `// Ensure the match will fit in the remainder of the span.` would suffice.

Reply:
Sure, I can add a comment.  FWIW I personally think it's easier to read with the addition rather than subtraction (the cast is a bit kludgy, though).